### ISSUE TITLE
Improve JWT consumer documentation for JWE scenarios

### ISSUE THREAD

**Issue by ayoho** [OpenLiberty/open-liberty#29823]
Issues like https://github.com/OpenLiberty/open-liberty/issues/29809 could be avoided with some improvements to our docs.

- The `sslRef` metatype description says:
   > Specifies an ID of the SSL configuration that is used to connect to the OpenID Connect provider.

   That's not entirely true or helpful in the case where the JWT consumer is consuming a token built by a Liberty JWT builder (i.e. via a `<jwtBuilder>` or the `JwtBuilder` API).
- We could consider updating the `keyManagementKeyAlias` metatype description for the jwtConsumer to explain the key needs to be in the keystore referenced by the `sslRef` attribute.
- The https://openliberty.io/docs/latest/reference/feature/jwt-1.0.html page could probably use a new section for configuring JWT consumers and/or builders to work with JWEs.

**Comment by c-koell** [comment]
@ayoho maybe you can help me as i have one more question about JWT :-)
We use now JWE and for me it's not really clear how the encrypted part in the JWE looks like. I have not really found informations in the liberty documentation. 

I have read about JWE Tokens and the ability to nest a JWS in it to also check the signature beside the decryption.
So does Liberty create JWE with a nested JWS Token in it ?

Thanks for any information!!

**Comment by ayoho** [comment]
@c-koell Yes, Liberty creates a JWE with a nested JWS within it. The code that creates the JWE is in the [JweHelper.createJweString() method](https://github.com/OpenLiberty/open-liberty/blob/integration/dev/com.ibm.ws.security.jwt/src/com/ibm/ws/security/jwt/utils/JweHelper.java#L53). That method is called by [JwtCreator.createJwt()](https://github.com/OpenLiberty/open-liberty/blob/integration/dev/com.ibm.ws.security.jwt/src/com/ibm/ws/security/jwt/utils/JwtCreator.java#L95), where you'll see we always create a JWS to use as the payload in the case of creating JWEs.

**Comment by c-koell** [comment]
@ayoho thank you so much for the information. I must say, its really great how fast you get informations from the liberty project community !

One question is still present :-) We use the com.ibm.websphere.security.jwt.JwtConsumer from
```
<dependency>
	<groupId>io.openliberty.features</groupId>
	<artifactId>jwt-1.0</artifactId>
	<version>${server.version}</version>
	<scope>provided</scope>
	<type>pom</type>
</dependency>
```
to extract the claims from the jwe token. Is there a way to extract the nested jws token from the jwe token ? Our idea is to use JWE  Tokens and store them on the client side (Mobile App) and if we receive the JWE Token on the server side we would like to extract the JWS token and use it for subsequent REST Calls in our MicroService Infrastructure. It's easier to configure JWK URL's to validate the signature rather than distribute Keystores to decrypt the Token.

**Comment by Kiiv** [comment]
Hi,

We have encountered the same difficulties as c-koell to configure our jwtConsumer to read a JWE token and I came here to open the same issue about the documentation.

I don't really understand why there is a "keyStoreRef" attribute on the jwtBuilder and why we have to pass an sslRef to the jwtConsumer but it's probably how it is implemented that is required this "trick".

If the documentation was more clear, it could be enough ;)
